lib/sshutils/x11/display.go

```diff
@@ -118,20 +118,37 @@ func (d *Display) Listen() (XServerListener, error) {
 
 // xserverUnixSocket returns the display's associated unix socket.
 func (d *Display) unixSocket() (*net.UnixAddr, error) {
-	// For x11 unix domain sockets, the hostname must be "unix" or empty. In these cases
-	// we return the actual unix socket for the display "/tmp/.X11-unix/X<display_number>"
+	// If hostname is "unix" or empty, then the actual unix socket
+	// for the display is "/tmp/.X11-unix/X<display_number>"
 	if d.HostName == "unix" || d.HostName == "" {
 		sockName := filepath.Join(x11SockDir(), fmt.Sprintf("X%d", d.DisplayNumber))
 		return net.ResolveUnixAddr("unix", sockName)
 	}
+
+	// It's possible that the display is actually the full path
+	// to an open XServer socket, such as with xquartz on OSX:
+	// "/private/tmp/com.apple.com/launchd.xxx/org.xquartz.com:0"
+	if d.HostName[0] == '/' {
```

Comment:
Is DISPLAY guaranteed to be an absolute path, if it's a unix socket?

Reply:
I haven't found documentation about this, but based on the openssh code it looks like it has to be an absolute path ending with `:display_num` or `:display_num.screen_num`.